server/server/settings_util.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+from decouple import Config, RepositoryEnv, undefined
+
+PROD_ENV_FILE_NAME = ".env.prod"
+DEV_PUBLIC_ENV_FILE_NAME = ".env.dev-public"
+DEV_SECRET_ENV_FILE_NAME = ".env.dev-secret"
+
+
+def get_project_config(BASE_DIR):
+    try:
+        config = Config(
+            RepositoryEnv(os.path.join(BASE_DIR, PROD_ENV_FILE_NAME))
+        )
+    except:
+
+        def dev_config_merged(*args, default=undefined, **kwargs):
+            try:
+                return Config(
+                    RepositoryEnv(
+                        os.path.join(BASE_DIR, DEV_SECRET_ENV_FILE_NAME)
+                    )
+                )(*args, **kwargs)
+            except:
+                return Config(
+                    RepositoryEnv(
+                        os.path.join(BASE_DIR, DEV_PUBLIC_ENV_FILE_NAME)
+                    )
+                )(*args, default, **kwargs)
+
+        config = dev_config_merged
+
+    return config
```

Comment:
I trust the tests but I'm super confused, can configs be lazy? How come this function can return a `Config` instance OR a function that resolves to a `Config` instance? Can we add a comment?

Where is the merge happening? It seems we only use the public file if the secret one doesn't exist. Ideally I'd like to put values like `ALLOWED_HOSTS=*` in the public-dev and _only_ things like `GC_NOTIFY_KEY` in the secret-dev but it looks like I'd have to repeat everything from the public-dev in the secret-dev. Unless decouple magic is throwing a special exception when a value lookup fails? If so, can we use that exception subclass and add a corresponding comment?